### PR TITLE
Search result highlighting doesn't work if the words are not in order #5897

### LINKED ISSUE #5897: Search result highlighting doesn't work if the words are not in order
image
image

### PR DESCRIPTION
**Pull request by tarun14110**
Fixes #5897

### CODE REVIEW

**Comment by tarun14110** [comment]
Can someone please open this PR. I closed by mistake.

**Comment by damithc** [comment]
Do we have any tests for this feature? If not, can we add some?

**Comment by chowyb** [comment]
I glanced quickly through the current tests for AdminSearch, and it appears that there are no tests for highlighting, only whether a certain entry is present or not.

For the changes introduced in this PR, the UI now looks as follows:

1. Exact order without quotations
search1
Previous behaviour: highlighted, including whitespaces

2. Non-exact order without quotations
search2
Previous behaviour: not highlighted

3. Exact match with quotations:
search3
Previous behaviour: not highlighted

4. Special case: Unicode character match
search4
Previous behaviour: not highlighted

While the highlighting is still not completely ideal yet, the changes here may be sufficient to address the issue for now. @damithc should we add anything else in this PR?

**Comment by damithc** [comment]
Is it possible to add tests to prevent future regressions? I mean, is it technically feasible and worth the trouble?

**Comment by chowyb** [comment]
The easiest way would be to test for this via HTML comparisons (GodMode), although that isn't too robust a method to test for this.

Writing methods to test for highlighted text instead (by checking if they lie within a `<span class="highlight">` element) could work as well, but is largely impractical as such a method would have to search through the HTML for the given text (effectively equivalent to GodMode in terms of performance) before checking for the presence of the `span` element.

However, I acknowledge that I might be wrong about this, so tagging @wkurniawan07 here for additional input.

**Comment by chowyb** [comment]
As mentioned by @kanghj in a discussion earlier, the highlight feature is already partially tested via comparison with HTML files (GodMode) in InstructorSearchPageUiTest, although all the cases so far appear to search for only a single keyword. It may be possible to add similar tests there for the case of multiple keywords.

**Comment by damithc** [comment]
Perhaps modify that test to use multiple keywords out of order? If that works, chances are it will work for single keyword as well as keywords in order.

**Comment by chowyb** [comment]
Okay then, @tarun14110 you'll have to modify the search tests to use multiple keywords out of order here.

**Comment by tarun14110** [comment]
@chowyb ready for review again.

File: src/main/webapp/js/common.js [hunk 1]
```diff
@@ -775,7 +775,7 @@ function highlightSearchResult(searchKeyId, sectionToHighlight) {
     var searchKey = $(searchKeyId).val();
     // trim symbols around every word in the string
     var symbolTrimmedSearchKey = [];
-    $.each(searchKey.split(/["'.-]/), function() {
+    $.each(searchKey.split(/[ "'.-]/), function() {
         symbolTrimmedSearchKey.push($.trim(this));
```
Comment (chowyb) [inline src/main/webapp/js/common.js]: Maybe rename this variable (and change the comment above) as this no longer applies to symbols only.

**Comment by chowyb** [comment]
One minor change requested.

**Comment by tarun14110** [comment]
@chowyb updated, please have a look.

**Pull request merged**